cloud/file-store/src/FileRoutes.ts

```diff
@@ -0,0 +1,71 @@
+import { Router } from 'express';
+import uuid from 'uuid-js';
+import { FileStorage } from '../file-api/FileStorage';
+import fileService from './services/File';
+
+/**
+ * Create express based router router for fileService
+ *
+ * @param {FileStorage} storageEngine - engine used to store files
+ * @returns router instance of express router
+ */
+function initRouter(storageEngine: FileStorage) {
+  const router = Router();
+
+  router.route('/:filename').post(function(req, res, next) {
+    const id = uuid.create().toString();
+    const fileMeta = {
+      owner: req.params.owner,
+      name: req.params.filename,
+      namespace: req.params.namespace,
+      id
+    };
+    const stream = fileService.parseBase64Stream(req);
+    fileService.writeStreamToFile(fileMeta, stream).then(function() {
+      const location = fileService.filePath(fileMeta.id);
+      return storageEngine.writeFile(fileMeta.namespace, fileMeta.id, location);
+    }).then(function() {
+      res.json(fileMeta);
+    }).catch(function(err) {
+      next(err);
+    });
+  });
+
+  const binaryUploadInitMiddleware = function(req, res, next) {
+    req.fileMeta = {};
+    req.fileMeta.id = uuid.create().toString();
+    req.fileMeta.uid = req.fileMeta.id;
+    req.fileMeta.name = req.body.fileName;
+    req.fileMeta.namespace = req.body.namespace;
+    req.fileMeta.owner = req.body.ownerId;
+    req.fileMeta.mimetype = req.file.mimetype;
+    next();
+  };
+
+  router.route('/:filename/binary').post(binaryUploadInitMiddleware, fileService.mutlerMiddleware,
```

Comment:
Do you think we need these separate endpoints? I think we can use http headers instead.
Seems like multer already only works on top of `multipart/form-data`, so for base64 we could go with `Content-Type: application/octet-stream` and maybe `Content-Transfer-Encoding: base4`, or even `Content-Type: application/base64`

See: https://www.w3.org/Protocols/rfc1341/5_Content-Transfer-Encoding.html

Reply:
Awesome comment. Fully agree. Going to merge them together!